AK/Coroutine.h

```diff
@@ -252,9 +252,9 @@ struct TryAwaiter {
 
 namespace Detail {
 template<typename T>
-auto declval_coro_result(Coroutine<T>&&) -> T;
+auto declval_coro_result(Coroutine<T> const&) -> T;
 template<typename T>
-auto declval_coro_result(T&&) -> T;
+auto declval_coro_result(T const&) -> T;
```

Comment:
(Not a review, obviously)
Any particular reason to change one flawed definition to another?

By "flawed" I mean that last two static asserts fail here:
```cpp
template<typename T>
struct F {
    using Type = T;

    F(T&) { }
    F(T&&) { }
};

template<typename T>
T my_decltype(T&&);

template<typename T>
T your_decltype(T const&);

static_assert(SameAs<int const, decltype(F(declval<int const&>()))::Type>);
static_assert(SameAs<int const, decltype(my_decltype(declval<int const&>()))>);
static_assert(SameAs<int const, decltype(your_decltype(declval<int const&>()))>);
```

Reply:
nah, I think this is just leftovers from the previous impl, I just didn't remove it when rebasing on master.